### ISSUE TITLE
Role of the "projection" layer

### ISSUE THREAD

**Issue by ReubenDo** [vlievin/biva-pytorch#3]
Hello,

First congratulations for the quality of your work and the clarity of the released code! It is a pleasure to work on it.

I was wondering, what is the underlying justification of the use of a fully-connected "projection layers" after sampling $z^{bu}_{i}$ and $z^{td}_{i}$? Is it only to model covariance structure?

Thank you,
Reuben

**Comment by vlievin** [comment]
Thank you Reuben. In the generative model, at each layer $i$ is sampled one latent variable $z_i = [z^{bu}_i, z^{td}_i]$. Depending on the architecture (MNIST/CIFAR), **z** is either (A) a stochastic feature map of shape [C, H, W] (for CIFAR) or (B) a simple latent of shape [C]. In both cases, the variable is fed to a convolutional NN, that accepts feature maps of shape [C, H, W]. 

- A (CIFAR): the variable **z** is already a feature map, hence compatible with the following conv layer. no projection layer is required. 
- B (MNIST): **z** is of shape [C] and needs to be plugged into the next conv layer. Multiple projections schemes are possible (fully connected linear/non-linear, expand the sample,...). Using a linear layer is a straightforward solution that also allows modelling the spatial structure (as the linear layer connects the global **z** of shape [C] to the map [C, H, W]). The CIFAR architecture can be used for MNIST however global latent variables (i.e. of shape [C]) allows modelling the sparse structure of MNIST more explicitly. 

The projection for cases A/B is handled by the class `asFeatureMap`.

In short, the fully-connected projection layer is the simplest solution to connect the global latent variable of shape [C] to the convolutional feature maps of shape [C, H, W].

**Comment by ReubenDo** [comment]
Great thank you for your answer!
